Automattic/jetpack
# Payment blocks: Extract Gutenberg context resolver to a separate component

Fixes #23826

#### Changes proposed in this Pull Request:

* Extracted the logic that determines the Gutenberg context to a separate component.
* Updated usages.
* Added tests for the component.

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Create or edit a Post
* Add a Payment Button and/or a Premium Content block.
* Click on the "Add a new payment plan" link on the main toolbar.
* If the sidebar has the add a new plan form it should now be focused.
* Repeat the process in the site editor and widget editor, the behavior should be the same.

#### Test evidence:

https://user-images.githubusercontent.com/1989914/162028784-26b3ba87-2ae7-45df-9a42-9644cc995119.mp4
